public/gpudevice.h

```diff
@@ -64,6 +64,12 @@ class GpuDevice : public HWCThread {
 
   void GetConnectedPhysicalDisplays(std::vector<NativeDisplay*>& displays);
 
+  bool IsSameResolutionForAllDisplay();
+
+  bool IsRotateMode() {
+    return rotate_mode_;
```

Comment:
What is rotate_mode_ mean?
each display can have different roate, for example, display 0 is not rotated, while display 1 is rotated

Reply:
Here it seems you make it golobal?

Reply:
I didn't check the rotate configuring for each display, currently, only check if the configure item ROTATION="false" is true or false.  Yes, I agree with you. we can always enable surface_damage for primary display and un-rotated display even with the exception. I will try to update it

Reply:
Why when rotated, surface damage can't be use?

Reply:
SF limitation, the coordinate of surface_damage/displayframe are calculated only with the primary display size. SF did not send the rotated coordinate to hwc when the extend display is rotated. HWC will use incorrect coordinate to set GL scissor. then some layer are not updated correctly

Reply:
The coordinate of surface_damage is not related with display size or rotation, that's my knowledge, the coordination of surface_damage is similar with source_corp as I know